app/src/main/java/com/github/bgabriel998/softwaredevproject/GalleryActivity.java

```diff
@@ -27,35 +29,37 @@ protected void onCreate(Bundle savedInstanceState) {
 
     /**
      * Gets the recycler view and fills it up with all images.
-     * TODO images list should be paths instead.
      */
     private void setupGallery(){
-        RecyclerView recyclerView = findViewById(R.id.gallery_recyclerview);
-
-        List<Integer> images = getImages();
+        List<String> imagePaths = getImagePaths();
+        if (!imagePaths.isEmpty()) {
+            findViewById(R.id.gallery_empty).setVisibility(View.GONE);
+        }
 
-        GalleryAdapter galleryAdapter = new GalleryAdapter(this, images, image -> {
+        GalleryAdapter galleryAdapter = new GalleryAdapter(this, imagePaths, imagePath -> {
             Intent intent = new Intent(this, ImageActivity.class);
-            intent.putExtra("image", image);
+            intent.putExtra("imagePath", imagePath);
```

Comment:
For the same reason as the last comment, would it be convenient to make "imagePath" a final static attribute? This way, we could call the same attribute when the extra of the intent is used instead of writing the same string two times